packages/pigeon/lib/generator_tools.dart

```diff
@@ -144,6 +144,13 @@ class Indent {
   void add(String text) {
     _sink.write(text);
   }
+
```

Comment:
I can't leave this comment on the right line since it's outside of the change context, but while we are improving this class to make the code to manage getting the right whitespace more readable, could you rename `scoped` to `addScoped`? Then we'd have `add`/`write`, `addln`/`writeln`, and `addScoped`/`writeScoped` as the same-behavior pairs, instead of `scoped`/`writeScoped`. The inconsistency there makes it hard to tell what `scoped` does, and thus easier to use the wrong one.

Reply:
I can do that.

I've been thinking about the naming conventions of scoped as well. Specifically the fact that people seem to miss the optional addTrailingNewline option often.

I was also considering creating a method for "indent to current level" method. And/or something to assist with looped scoped statements.